### PR TITLE
fix: components should only react to primary button pointer events.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #5513

## Summary

fix: components should only react to primary button pointer events.

This PR checks that the button is `0` for `pointerup` and `pointerdown` events throughout the components.

https://www.w3.org/TR/pointerevents/#the-button-property

image

### CODE REVIEW

**Comment by driskull** [comment]
@jcfranco should we also have a check for `pointerup` events?

File: src/utils/dom.ts [hunk 1]
```diff
@@ -289,3 +289,15 @@ export function intersects(rect1: DOMRect, rect2: DOMRect): boolean {
...
+
+/**
+ * This helper returns true if the pointer event fired from the primary button of the device.
+ *
+ * See https://www.w3.org/TR/pointerevents/#the-button-property.
+ *
+ * @param event
+ * @returns {boolean}
+ */
+export function isPrimaryPointerButton(event: PointerEvent): boolean {
```
Comment (jcfranco) [inline src/utils/dom.ts:301]: Can you add a spec test for this?

File: src/components/slider/slider.tsx [hunk 2]
```diff
@@ -989,12 +993,16 @@ export class Slider
     return ticks;
   }
 
-  private dragStart(prop: ActiveSliderProperty): void {
```
Comment (jcfranco) [inline src/components/slider/slider.tsx:992]: Nitpick: for consistency, can you bring this back and use it in `pointerDownDragStart` (similar to `pointerDownDragEnd`)?

**Review by jcfranco — APPROVED** [review]
Thanks for fixing this! 🐁👈

Side bar: we should revisit how we're using pointer events as clicks/taps. `pointerdown` & `click` fire at different times and with different context (e.g., click has a slight delay before firing, and `mousedown` & `mouseup` needs to fire from the same element), so this could throw some users off. Maybe we could bring back listening to `click` and for the rest, we can rely on pointer events.

File: src/utils/dom.ts (hunk 1 above)
Comment (driskull) [reply src/utils/dom.ts:301]: Yesssir \( ﾟヮﾟ)/

File: src/components/slider/slider.tsx (hunk 2 above)
Comment (driskull) [reply src/components/slider/slider.tsx:992]: done! (*・‿・)ノ⌒*:･ﾟ✧

**Review by jcfranco — APPROVED** [review]
💥

**Pull request merged**